### PR TITLE
Add confirmation dialog box when saving and file already exists

### PR DESCRIPTION
**Pull request by Quasar985**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Added dialog box when saving a graph that appears when file already exists.

image

### Alternate Designs

N/A

### Why Should This Be In Core?

N/A

### Benefits

User's will now be less likely to mistakenly overwrite files.

### Possible Drawbacks

None.

### Verification Process

Automated testing and manual testing.

### Applicable Issues

https://github.com/constellation-app/constellation/issues/1957

### CODE REVIEW

**Review by Delphinus8821 — APPROVED** [review]

**Review by capricornunicorn123 — COMMENTED** [review]
The changes look good. Just as an FYI, there 2 main Implementations of saving in Constellation (that I know of): 

1. Saving through the help of PluginParameters (See FileInputPane.GetFileChooser(...))
   - This effects some export plugins such as SVG Export and geo exports
2. Local Implementation of saving. see ExportToJsonAction.actionPerformed(...) & ExportToImage.actionPerformed(...)
   - This is implemented in a duplicated way across a number other export plugins, similar to the implementation the class you have already changed.

What are your thoughts on localizing the Save/Open dialog FileChooserBuilder construction to a central utility class? The dialog opening functionality has been centralized to the FileChooser class. Dialog building functionality could also be included in this class with all of the individual uses of save/open dialogs calling methods in the FileChooser class.

**Comment by Quasar985** [comment]
> The changes look good. Just as an FYI, there 2 main Implementations of saving in Constellation (that I know of):
> 
> 1. Saving through the help of PluginParameters (See FileInputPane.getFileChooser(...))
>    
>    * This effects some export plugins such as svg Export and geo exports
> 2. Local Implementation of saving. see ExportToJsonAction.actionPerformed(...) & ExportToImage.actionPerformed(...)
>    
>    * This is implemented in a duplicated way across a number other export plugins, similar to the implementation the class you have already changed.
> 
> What are your thoughts on localizing the Save/Open dialog FileChooserBuilder construction to a central utility class? The dialog opening functionality has been centralized to the FileChooser class. the building functionality can also be moved here or to a new class. All of the individual uses of save/open dialogs can then call methods from one place to improve consistency.

I'll have to take a look at how that all works. It would make sense to have these duplicated functions be centralized if possible.

**Comment by mimosa2** [comment]
I tested 'save as' from file menu for .star types it displayed dialog box correctly as expected. When I tried Export -> To JSON then it still overwrites (same for other types). May be a check for other types missing?

**Comment by Quasar985** [comment]
> I tested 'save as' from file menu for .star types it displayed dialog box correctly as expected. When I tried Export -> To JSON then it still overwrites (same for other types). May be a check for other types missing?

It looks like the file I changed only covers saving the graph itself. I'll look into fixing the problem for all types of exporting. I reckon I'll convert this PR back to draft in the meantime.

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/SaveAsAction.java [hunk 1]
```diff
@@ -312,21 +311,39 @@ boolean _isEnabled() {
...
-                        return (file.isFile() && StringUtils.endsWithIgnoreCase(name, FileExtensionConstants.STAR)) || file.isDirectory();
-                    }
-
-                    @Override
-                    public String getDescription() {
-                        return "Constellation Files (" + FileExtensionConstants.STAR + ")";
-                    }
-                });
+        return FileChooser.createFileChooserBuilder(TITLE, FileExtensionConstants.STAR, "Constellation Files (" + FileExtensionConstants.STAR + ")");
+//        return new FileChooserBuilder(TITLE)
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/SaveAsAction.java:315]: can this commented out but be removed if no longer needed?

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**